src/components/frame/v5/pages/AgreementsPage/hooks.ts

```diff
@@ -15,23 +15,55 @@ import { notNull } from '~utils/arrays/index.ts';
 
 import { useFiltersContext } from './FiltersContext/FiltersContext.ts';
 
-export const useGetAgreements = () => {
+const QUERY_PAGE_SIZE = 20;
+
+const useGetAllAgreements = () => {
   const {
     colony: { colonyAddress },
   } = useColonyContext();
-  const { activeFilters, searchFilter } = useFiltersContext();
-  const { data, loading } = useGetColonyActionsQuery({
+
+  const { data, loading, fetchMore } = useGetColonyActionsQuery({
     variables: {
       colonyAddress,
       filter: {
         type: { eq: ColonyActionType.CreateDecisionMotion },
       },
       sortDirection: ModelSortDirection.Desc,
+      limit: QUERY_PAGE_SIZE,
     },
     fetchPolicy: 'network-only',
   });
 
   const agreementsData = data?.getActionsByColony?.items.filter(notNull);
+  const nextToken = data?.getActionsByColony?.nextToken;
+
+  const fetchNextPage = useCallback(
+    async (token) => {
+      await fetchMore({
+        variables: { nextToken: token },
+      });
+    },
+    [fetchMore],
+  );
+
+  useEffect(() => {
+    if (nextToken) {
+      fetchNextPage(nextToken);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [nextToken]);
```

Comment:
There's a better way to do this, without `useEffect`. See this implementation: https://github.com/JoinColony/colonyCDapp/blob/master/src/context/TokenSelectContext/TokenSelectContextProvider.tsx#L27-L52

Reply:
That is quite a nice implementation, thanks @rdig  🚀 On to updating mine